devnotes/tpm.md

```diff
@@ -0,0 +1,258 @@
+# TPM for IBM POWER9 - feasibility study
+
+This document is an attempt to gather information about the issues faced with
+TPMs on POWER9 and what are the potential solutions to them.
+
+Mind that thanks to POWER9 being more secure than any x86 by design, it is more
+important to have TPM support than to have a particularly secure or feature-full
+one.  See [platform comparison] for more details.
+
+## Some definitions
+
+* TPM - Trusted Platform Module
+* dTPM - Discreet TPM module (a dedicated pluggable hardware module)
+* fTPM - Firmware-based TPM implementation
+* LPC - Low Pin Count (hardware interface)
+* I2C - Inter-Integrated Circuit (hardware interface)
+* SBE - Self-Boot Engine (part of POWER9 CPU and/or its firmware)
+* MCU - MicroController Unit (like a board connected via USB)
+* FPGA - Field-Programmable Gate Array (programmable integrated circuit)
+* BMC - Baseboard Management Controller ("service processor")
+
+## Problems
+
+### Talos II cannot generate LPC TPM cycles
+
+This is a hardware limitation and cannot be worked around in firmware.
+
+The consequence is that regular LPC TPM cannot be used on Talos II by directly
+connecting it to the TPM LPC header (one might wonder why have the LPC TPM
+header in the first place...).
+
+## Quick overview
+
+| Aspect\Approach | [I2C]        | [On BMC] | [IO LPC]   | [FPGA SPI] | [On MCU]      | [LibreBMC]    | [SBE]    |
+| --------------- | -----        | -------- | --------   | ---------- | --------      | ----------    | -----    |
+| Interface       | I2C          | -        | LPC        | SPI        | I2C           | ?             | -        |
+| TPM kind        | dTPM         | fTPM     | dTPM       | dTPM       | dTPM          | ?             | -        |
```

Comment:
SBE is definitely fTPM

LibreBMC I would call dTPM, since it is externally connected component. It relies on software but all TPM does. I'm not sure if strict enough definition of dTPM vs fTPM exist. At some point we published blog about that: https://blog.3mdeb.com/2021/2021-10-08-ftpm-vs-dtpm/